domain/todo_usecase_interface.go

```diff
@@ -8,4 +8,6 @@ import (
 type TodoUseCase interface {
 	CreateTodoUseCase(ctx context.Context, entity CreateTodoInputEntity) (result *CreateTodoEntity, err error)
 	GetTodoUseCase(ctx context.Context, key *uuid.UUID) (result []GetTodoEntity, err error)
+
+	UpdateTodoUseCase(ctx context.Context, queryEntity UpdateTodoQueryEntity, entity *UpdateTodoEntity) (result *UpdateTodoEntity, err error)
```

Comment:
UpdateTodoEntity should not be pointer, it will your update value
UpdateTodoQueryEntity can be pointer (incase of you want to update all)